src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
> Can you please elaborate why exactly you need laziness? Is it because you set the alignment after the block is crated? Will this help?

I need to read user data when alignment for (first) child is created, not when `getAlignmentStrategy` is fired, because there is no guarantee that something won't change between `getAlignmentStrategy` and children building loop.

I'm aware that this design is ugly, and I'm working on replacing this whole mess with separate formatting blocks for _special_ PSI nodes. Unfortunately I don't have a lot of spare time currently (finals are coming), so this may wait for a bit.